File: Problem-042/solution.js
function codedTriangleNumbers(n) {
    let count = 0;

    for (let i = 0; i < n; i++) {
        let word = words[i];
        let score = 0;
        for (let char of word) {
            score += char.charCodeAt(0) - 64;
        }
        if (isTriangle(score)) {
            count++;
        }
    }

    return count;
}

function isTriangle(num) {
    let n = 1;
    let triangle = 1;

    while (triangle <= num) {
        if (triangle === num) {
            return true;
        }

        n++;
        triangle = n * (n + 1) / 2;
    }

    return false;
}

File: Problem-042/solution.py
def coded_triangle_numbers(n):
    count = 0

    for i in range(n):
        word = words[i]
        score = 0

        for char in word:
            score += ord(char) - ord('A') + 1

        if is_triangle(score):
            count += 1

    return count


def is_triangle(num):
    n = 1
    triangle = 1

    while triangle <= num:
        if triangle == num:
            return True

        n += 1
        triangle = n * (n + 1) // 2

    return False
